main.go

```
@@ -0,0 +1,83 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"net"
+	"strconv"
+	"strings"
+)
+
+func TCPServer() {
+	l, err := net.Listen("tcp", ":8081")
+	if err != nil {
+		log.Fatal(err)
+	}
+	fmt.Println("(S): listening on :8081")
+
+	for {
+		conIn, err := l.Accept()
+		if err != nil {
+			log.Fatal(err)
+		}
+		fmt.Println("(S): connection from ", conIn.RemoteAddr().String())
+		for {
+			buf := make([]byte, 1024)
+			n, err := conIn.Read(buf)
+			if err != nil {
+				log.Fatal(err)
+			}
+
+			m := string(buf[0:n])
+			if i, err := strconv.ParseInt(m, 10, 64); err == nil {
+				fmt.Println("(S):I ", fmt.Sprintf("%d", i*2))
+				conIn.Write([]byte(fmt.Sprintf("%d", i*2)))
+			} else if f, err := strconv.ParseFloat(m, 64); err == nil {
+				fmt.Println("(S):F ", fmt.Sprintf("%f", f))
+				conIn.Write([]byte(fmt.Sprintf("%f", f)))
+			} else {
+				fmt.Println("(S):str ", m)
+				if m == "exit" {
+					conIn.Close()
+					fmt.Println("(S): ", m)
+					break
+				}
+				conIn.Write([]byte(strings.ToUpper(m)))
+			}
+		}
+	}
+
+}
+
+func TCPClient() {

```

I would recommend to return an error from this function instead of log.Fatal